src/domTree.js

```diff
@@ -25,6 +28,103 @@ const createClass = function(classes: string[]): string {
     return classes.filter(cls => cls).join(" ");
 };
 
+const initNode = function(
+    classes?: string[],
+    options?: Options,
+    style?: CssStyle,
+) {
+    this.classes = classes || [];
+    this.attributes = {};
+    this.height = 0;
+    this.depth = 0;
+    this.maxFontSize = 0;
+    this.style = style || {};
+    if (options) {
+        if (options.style.isTight()) {
+            this.classes.push("mtight");
+        }
+        const color = options.getColor();
+        if (color) {
+            this.style.color = color;
+        }
+    }
+};
+
+/**
+ * Convert into an HTML node
+ */
+const toNode = function(tagName: string): HTMLElement {
+    const node = document.createElement(tagName);
+
+    // Apply the class
+    node.className = createClass(this.classes);
+
+    // Apply inline styles
+    for (const style in this.style) {
+        if (this.style.hasOwnProperty(style)) {
```

Comment:
Maybe we should go with
```
Object.keys(this.style).forEach(
   key => node.style[key] = this.style[key]);
```
It's faster and short.

Reply:
@kevinbarabash I agree functional approach is the best. But in that case, it uses `Object.keys()` which this PR aims to avoid. I'll open another PR, changing loops to use map, every, filter, find, findIndex, reduce, some, etc.